x-pack/plugins/spaces/server/routes/api/public/delete.ts

```diff
@@ -7,16 +7,17 @@
 import Boom from 'boom';
 import { wrapError } from '../../../lib/errors';
 import { SpacesClient } from '../../../lib/spaces_client';
+import { PublicRouteDeps, PublicRouteRequestFacade } from '.';
 
-export function initDeleteSpacesApi(server: any, routePreCheckLicenseFn: any) {
-  server.route({
+export function initDeleteSpacesApi(deps: PublicRouteDeps) {
+  const { http, savedObjects, spacesService, routePreCheckLicenseFn } = deps;
+
+  http.route({
     method: 'DELETE',
     path: '/api/spaces/space/{id}',
-    async handler(request: any, h: any) {
-      const { SavedObjectsClient } = server.savedObjects;
-      const spacesClient: SpacesClient = server.plugins.spaces.spacesClient.getScopedClient(
-        request
-      );
+    async handler(request: PublicRouteRequestFacade, h: any) {
+      const { SavedObjectsClient } = savedObjects;
+      const spacesClient: SpacesClient = spacesService.scopedClient(request);
```

Comment:
TODO for myself: either we refactor `SpacesClient` to work with `KibanaRequest` or we should provide raw hapi request in router handler as well for transition period